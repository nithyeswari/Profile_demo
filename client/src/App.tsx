 import UserSummary from './features/summary/UserSummary';

function App() {
  return (
    <div>
      <UserSummary>
      </UserSummary> 
    </div> 
  );
}

export default App;
