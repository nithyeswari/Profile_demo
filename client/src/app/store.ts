import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import { createStore, combineReducers } from 'redux'
import { reducer as formReducer } from 'redux-form'

const reducers = {
  form: formReducer
}

const reducer = combineReducers(reducers)

export const store = createStore(reducer)
 

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
