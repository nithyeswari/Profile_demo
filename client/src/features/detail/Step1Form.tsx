import React from "react";
import { Field, reduxForm, InjectedFormProps } from "redux-form";
import CustomInput from '../shared/CustomInput';
import IFormProps from "../model/IFormProps";
import Validate from '../validations/Validate';
import Card from '@material-ui/core/Card'; 
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
    root: {
        '& .MuiTextField-root': {
            margin: theme.spacing(1),
            width: 200,
        },
    },
}));
let Step1Form: React.FC<IFormProps & InjectedFormProps<{}, IFormProps>> = (props: any) => {
    const classes = useStyles();
    const { handleSubmit, pristine, reset, submitting } = props

    return (
        <Card>
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    Step 1
                </Typography>
                <form id="detail" className={classes.root} onSubmit={handleSubmit} autoComplete="off">
                    <Field name='fullname'
                        component={CustomInput}
                        label='Full name'
                        type='text'></Field>
                    <Field
                        name="email"
                        label='Email'
                        component={CustomInput}
                        type="email"
                        placeholder="Email"
                    />   <Field
                        name="telephone"
                        label='Telephone'
                        component={CustomInput}
                        type="phone"
                        placeholder="Phone number"
                    />
                </form>
            </CardContent>

            <CardActions>
                <Button size="small" color="primary" type="submit" disabled={pristine || submitting}  form="detail"   >
                    Next
                </Button>
                <Button size="small" color="primary" type="button" disabled={pristine || submitting} onClick={reset}>
                    Clear
                </Button>
            </CardActions>
        </Card>


    )
}
export default reduxForm<{}, IFormProps>({
    form: 'detail',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate: Validate
})(Step1Form)
