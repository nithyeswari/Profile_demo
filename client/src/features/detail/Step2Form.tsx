import React from "react";
import { FieldArray, reduxForm, InjectedFormProps } from "redux-form";
import { CustomFieldArray } from "../shared/CustomeFieldArray";
import IFormProps from "../model/IFormProps";
import Validate from '../validations/Validate';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

let Step2Form: React.FC<IFormProps & InjectedFormProps<{}, IFormProps>> = (props: any) => {
    const { handleSubmit, pristine, reset, submitting, prevPage } = props
    return (
        <Card>
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    Step 2
                </Typography>
                <form id="detail" onSubmit={handleSubmit}  autoComplete="off">
                    <div>
                        <FieldArray name="skills" component={CustomFieldArray} ></FieldArray>
                    </div>
                </form>
            </CardContent>

            <CardActions>
                <Button size="small" color="primary" type="button" onClick={prevPage} >
                    Prev
                </Button>
                <Button size="small" color="primary" type="submit" disabled={pristine || submitting} form="detail"   >
                    Next
                </Button>
                <Button size="small" color="primary" type="button" disabled={pristine || submitting} onClick={reset}>
                    Clear
                </Button>
            </CardActions>
        </Card> 

    )
}
export default reduxForm<{}, IFormProps>({
    form: 'detail',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate: Validate
})(Step2Form)
