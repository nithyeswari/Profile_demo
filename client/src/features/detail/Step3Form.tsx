import React, { Component } from "react";
import { Field, InjectedFormProps, reduxForm } from "redux-form";
import CustomInput from '../shared/CustomInput';
import IFormProps from "../model/IFormProps";
import Validate from '../validations/Validate';
import CustomCheckbox from "../shared/CustomCheckbox";

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

let Step3Form: React.FC<IFormProps & InjectedFormProps<{}, IFormProps>> = (props: any) => {
    const { handleSubmit, pristine, reset, submitting, prevPage } = props
    return (

        <Card>
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    Step 3
                </Typography>
                <form id="detail" onSubmit={handleSubmit}  autoComplete="off">
                    <div>
                        <label htmlFor="canText">Send feedback via text </label>
                        <div>
                            <Field
                                name="canText"
                                id="canText"
                                component={CustomCheckbox}
                                type="checkbox"
                            />
                        </div>
                    </div>

                    <div>
                        <label htmlFor="canEmail">Send feedback via email</label>
                        <div>
                            <Field
                                name="canEmail"
                                id="canEmail"
                                component={CustomCheckbox}
                                type="checkbox"
                            />
                        </div>
                    </div>

                    <div>
                        <label htmlFor="isTc">Accept T&C </label>
                        <div>
                            <Field
                                name="isTc"
                                id="isTc"
                                component={CustomCheckbox}
                                type="checkbox"
                            />
                        </div>
                    </div>

                     
                </form>
            </CardContent>

            <CardActions>
                <Button size="small" color="primary" type="button" onClick={prevPage} >
                    Prev
                </Button>
                <Button size="small" color="primary" type="button" disabled={pristine || submitting} onClick={reset}>
                    Clear
                </Button>
                <Button size="small" color="primary" type="submit" disabled={pristine || submitting} form="detail"   >
                    Save
                </Button>
               
            </CardActions>
        </Card>




    )
}
export default reduxForm<{}, IFormProps>({
    form: 'detail',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate: Validate
})(Step3Form)
