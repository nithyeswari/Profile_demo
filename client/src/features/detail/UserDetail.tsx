

import { useState } from "react";
import Step1Form from "./Step1Form";
import Step2Form from "./Step2Form";
import Step3Form from "./Step3Form";
const UserDetail = (props: { onSubmit: any }) => {
    const { onSubmit } = props
    const [page, setPage] = useState(1);
    
    return (

        <div> 
            {page === 1 &&
                <Step1Form
                    handleSubmit={(data: any) => {
                        console.log(data);
                        setPage(2)
                    }} />
            }
            {page === 2 &&
                <Step2Form
                    handleSubmit={(data: any) => {
                        console.log(data);
                        setPage(3)
                    }}
                    prevPage={() => setPage(1)}
                />
            }
            {page === 3 &&
                <Step3Form
                    handleSubmit={onSubmit}
                    prevPage={() => setPage(2)}
                />
            }
        </div>
    )

};


export default UserDetail;