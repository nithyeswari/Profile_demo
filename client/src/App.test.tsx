import { render ,screen} from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';
import userEvent from '@testing-library/user-event';
import UserDetail from './features/detail/UserDetail';
import Step1Form from './features/detail/Step1Form';
import Step2Form from './features/detail/Step2Form';
import Step3Form from './features/detail/Step3Form';

test('Shows dummy profile data', () => {
  const { container } =render(
    <Provider store={store}>
      <App />
    </Provider>
  ); 
  expect(container.firstChild).toMatchSnapshot();
});
test('Able to edit summy profile data', () => {
   render(
    <Provider store={store}>
      <App />
    </Provider>
  ); 
  expect(screen.queryByText('Edit')).toBeInTheDocument(); 
  userEvent.click(screen.getByRole('button', { name: /Edit/i }));
  expect(screen.getByText('Step 1')).toBeInTheDocument();
});
test('Able to edit and save Step1', () => {
  const { container } = render(
   <Provider store={store}>
       <Step1Form handleSubmit={() => {}} prevPage={()=>{}}></Step1Form>
   </Provider>
 ); 
 expect(screen.queryByText('Next')).toBeInTheDocument(); 
 expect(screen.queryByText('Clear')).toBeInTheDocument();  
 expect(container.firstChild).toMatchSnapshot();
});
test('Able to edit and save Step2', () => {
  const { container } = render(
   <Provider store={store}>
     <Step2Form handleSubmit={() => {}} prevPage={()=>{}}></Step2Form>
   </Provider>
 ); 
 expect(screen.queryByText('Next')).toBeInTheDocument(); 
 expect(screen.queryByText('Clear')).toBeInTheDocument(); 
 expect(screen.queryByText('Prev')).toBeInTheDocument();  
 expect(container.firstChild).toMatchSnapshot();
});

test('Able to edit and save Step3', () => {
  const { container } = render(
   <Provider store={store}>
     <Step3Form handleSubmit={() => {}} prevPage={()=>{}}></Step3Form>
   </Provider>
 );  
 expect(screen.queryByText('Clear')).toBeInTheDocument(); 
 expect(screen.queryByText('Prev')).toBeInTheDocument();  
 expect(screen.queryByText('Save')).toBeInTheDocument();  
 expect(container.firstChild).toMatchSnapshot();
});