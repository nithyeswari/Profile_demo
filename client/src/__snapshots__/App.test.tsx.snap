// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Able to edit and save Step1 1`] = `undefined`;

exports[`Able to edit and save Step1 2`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Step 1
      </h2>
      <form
        autocomplete="off"
        class="makeStyles-root-15"
        id="detail"
      >
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Full name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                name="fullname"
                placeholder="Full name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-18"
                >
                  <span>
                    Full name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-18"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
           
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
            >
              Telephone
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                name="telephone"
                placeholder="Telephone"
                type="phone"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-18"
                >
                  <span>
                    Telephone
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
        disabled=""
        form="detail"
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Clear
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Able to edit and save Step2 1`] = `
<div
  class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Step 2
    </h2>
    <form
      autocomplete="off"
      id="detail"
    >
      <div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add 
              Skills
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Prev
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled=""
      form="detail"
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Clear
      </span>
    </button>
  </div>
</div>
`;

exports[`Able to edit and save Step3 1`] = `
<div
  class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Step 2
    </h2>
    <form
      autocomplete="off"
      id="detail"
    >
      <div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add 
              Skills
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Prev
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled=""
      form="detail"
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Clear
      </span>
    </button>
  </div>
</div>
`;

exports[`Shows dummy profile data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Your full name
        </h2>
        <ul
          class="MuiList-root makeStyles-root-1 MuiList-padding"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-.4 4.25l-7.07 4.42c-.32.2-.74.2-1.06 0L4.4 8.25c-.25-.16-.4-.43-.4-.72 0-.67.73-1.07 1.3-.72L12 11l6.7-4.19c.57-.35 1.3.05 1.3.72 0 .29-.15.56-.4.72z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Email
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Your email address
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2.5 20h-3c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h3c.28 0 .5.22.5.5s-.22.5-.5.5zm3.5-3H7V4h10v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Phone
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Your phone number
              </p>
            </div>
          </li>
        </ul>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="makeStyles-root-1"
        >
          <li>
            <div
              class="MuiChip-root makeStyles-chip-2"
            >
              <span
                class="MuiChip-label"
              >
                Angular
              </span>
            </div>
          </li>
          <li>
            <div
              class="MuiChip-root makeStyles-chip-2"
            >
              <span
                class="MuiChip-label"
              >
                jQuery
              </span>
            </div>
          </li>
          <li>
            <div
              class="MuiChip-root makeStyles-chip-2"
            >
              <span
                class="MuiChip-label"
              >
                Polymer
              </span>
            </div>
          </li>
          <li>
            <div
              class="MuiChip-root makeStyles-chip-2"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiChip-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                />
              </svg>
              <span
                class="MuiChip-label"
              >
                React
              </span>
            </div>
          </li>
          <li>
            <div
              class="MuiChip-root makeStyles-chip-2"
            >
              <span
                class="MuiChip-label"
              >
                Vue.js
              </span>
            </div>
          </li>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Edit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Share
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
